### PR TITLE
HW6 JDI Test Case

### PR DESCRIPTION
**Pull request by NikonovaEkaterina**
HW 6 JDI Test Case version 1

### CODE REVIEW

File: src/test/java/hw6/MetalAndColorsFormTest.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+
+import static hw6.JdiSite.*;
+
+public class MetalAndColorsFormTest extends AbstractJdiTest {
+
+
+    @Test(dataProviderClass = DataProviderTestData.class,
+            dataProvider = "getMetalsAndColorsTestData")
+    public void idiMetalAndColorsFormTest(int[] summary, String[] elements,
```
Comment (troe88) [inline src/test/java/hw6/MetalAndColorsFormTest.java]: Take a look at Entity Driving testing and Login form.
You need to create Site.Page.Form::submit(Entity) <- this structure.

File: src/main/java/hw6/pages/MetalAndColorsPage.java [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+    }
+
+    // Numbers summary
+    @UI("[name=custom_radio_odd]")
+    public RadioButtons oddNumbers;
+
+    @UI("[name=custom_radio_even]")
+    public RadioButtons evenNumbers;
+
+    public void selectNumbers(int[] numbers) {
```
Comment (troe88) [inline src/main/java/hw6/pages/MetalAndColorsPage.java]: This is not the best idea to create switch case, use WebList or any other JDI element, that allow you to find appropriate number.

File: src/main/java/hw6/pages/MetalAndColorsPage.java [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+import com.epam.jdi.light.elements.pageobjects.annotations.locators.UI;
+import com.epam.jdi.light.elements.pageobjects.annotations.locators.XPath;
+import com.epam.jdi.light.ui.html.elements.common.Button;
+import com.epam.jdi.light.ui.html.elements.common.Checkbox;
+import com.epam.jdi.light.ui.html.elements.common.Text;
+import com.epam.jdi.light.ui.html.elements.complex.RadioButtons;
+
+public class MetalAndColorsPage extends WebPage {
+
+    // Colors dropdown
```
Comment (troe88) [inline src/main/java/hw6/pages/MetalAndColorsPage.java]: You shold not live a comment like this. What is the reason of it ? 
All of us can realise that **colorsDropdown** field is **Colors dropdown**

File: src/main/java/hw6/pages/MetalAndColorsPage.java [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+        oddNumbers.select(first);
+        evenNumbers.select(second);
+    }
+
+
+    //Elements checklist
+    @FindBy(css = "section[id=elements-checklist] input[type=checkbox]")
+    public Checklist earthElements;
+
+    @XPath("//input[@id='g1']")
```
Comment (troe88) [inline src/main/java/hw6/pages/MetalAndColorsPage.java]: You have a earthElements Checklist, so what is the reason of checkboxes fields below ?
You need to get rid of them.

File: src/main/java/hw6/pages/MetalAndColorsPage.java [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+
+    //Vegetables dropdown with checboxes
+    @JDropdown(root = "div[ui=droplist]",
+            value = ".filter-option",
+            list = "li",
+            expand = ".caret")
+    public Dropdown vegetablesDropdown;
+
+    @XPath("//input[@id='g5']")
+    public Checkbox cucumberCheckBox;
```
Comment (troe88) [inline src/main/java/hw6/pages/MetalAndColorsPage.java]: Same story like elements, you already have vegetablesDropdown, this should be enough

File: src/main/java/hw6/pages/MetalAndColorsPage.java [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+    @UI("[id=submit-button]")
+    public Button submitButton;
+
+    public void clickSubmitButton() {
+        submitButton.click();
+    }
+
+    // Results section
+    @UI("[class=summ-res]")
+    public Text summaryResult;
```
Comment (troe88) [inline src/main/java/hw6/pages/MetalAndColorsPage.java]: You can create only one element like WebList and get collection of strings from it

**Review by troe88 — CHANGES_REQUESTED** [review]

File: src/main/java/hw6/pages/MetalAndColorsPage.java (hunk 6 above)
Comment (NikonovaEkaterina) [reply src/main/java/hw6/pages/MetalAndColorsPage.java]: Change to WebList

File: src/main/java/hw6/pages/MetalAndColorsPage.java (hunk 5 above)
Comment (NikonovaEkaterina) [reply src/main/java/hw6/pages/MetalAndColorsPage.java]: Removed

File: src/main/java/hw6/pages/MetalAndColorsPage.java (hunk 4 above)
Comment (NikonovaEkaterina) [reply src/main/java/hw6/pages/MetalAndColorsPage.java]: Removed

File: src/main/java/hw6/pages/MetalAndColorsPage.java (hunk 3 above)
Comment (NikonovaEkaterina) [reply src/main/java/hw6/pages/MetalAndColorsPage.java]: Removed comments like that

File: src/main/java/hw6/pages/MetalAndColorsPage.java (hunk 2 above)
Comment (NikonovaEkaterina) [reply src/main/java/hw6/pages/MetalAndColorsPage.java]: changed to another logic

File: src/test/java/hw6/MetalAndColorsFormTest.java (hunk 1 above)
Comment (NikonovaEkaterina) [reply src/test/java/hw6/MetalAndColorsFormTest.java]: Changed structure. Try to apply ED to the current implementation.

**Review by troe88 — APPROVED** [review]

**Pull request merged**